Spigot-Server-Patches/0703-ANSI-support-for-server-GUI.patch

```diff
@@ -0,0 +1,272 @@
+ 
+     private JComponent e() {
+         JPanel jpanel = new JPanel(new BorderLayout());
+-        JTextArea jtextarea = new JTextArea();
++        // Paper start - add support for ANSI
++        //JTextArea jtextarea = new JTextArea();
++        final JTextPane jtextarea;
++        // Respect terminal.ansi setting
++        if (ansiEnabled) {
++            jtextarea = new ANSITextPane();
++        } else {
++            jtextarea = new JTextPane();
++        }
++        // Paper end
+         JScrollPane jscrollpane = new JScrollPane(jtextarea, 22, 30);
+ 
+         jtextarea.setEditable(false);
+@@ -174,14 +186,14 @@ public class ServerGUI extends JComponent {
+         this.e.forEach(Runnable::run);
+     }
+ 
+-    private static final java.util.regex.Pattern ANSI = java.util.regex.Pattern.compile("\\x1B\\[([0-9]{1,2}(;[0-9]{1,2})*)?[m|K]"); // CraftBukkit
+-    public void a(JTextArea jtextarea, JScrollPane jscrollpane, String s) {
++    //private static final java.util.regex.Pattern ANSI = java.util.regex.Pattern.compile("\\x1B\\[([0-9]{1,2}(;[0-9]{1,2})*)?[m|K]"); // CraftBukkit // Paper - add support for ANSI
++    public void a(JTextPane jtextarea, JScrollPane jscrollpane, String s) { // Paper - add support for ANSI
+         if (!SwingUtilities.isEventDispatchThread()) {
+             SwingUtilities.invokeLater(() -> {
+                 this.a(jtextarea, jscrollpane, s);
+             });
+         } else {
+-            Document document = jtextarea.getDocument();
++            //Document document = jtextarea.getDocument(); // Paper - add support for ANSI
+             JScrollBar jscrollbar = jscrollpane.getVerticalScrollBar();
+             boolean flag = false;
+ 
+@@ -189,11 +201,24 @@ public class ServerGUI extends JComponent {
+                 flag = (double) jscrollbar.getValue() + jscrollbar.getSize().getHeight() + (double) (ServerGUI.a.getSize() * 4) > (double) jscrollbar.getMaximum();
+             }
+ 
+-            try {
+-                document.insertString(document.getLength(), ANSI.matcher(s).replaceAll(""), (AttributeSet) null); // CraftBukkit
+-            } catch (BadLocationException badlocationexception) {
+-                ;
++            // Paper start - add support for ANSI
++            // Respect terminal.ansi setting
++            if (ansiEnabled) {
++                if (jtextarea instanceof ANSITextPane) {
++                    final ANSITextPane ansiTextPane = (ANSITextPane) jtextarea;
++                    ansiTextPane.appendANSI(s);
++                } else {
++                    insertString(jtextarea.getDocument(), s);
++                }
++            } else {
++                insertString(jtextarea.getDocument(), s);
+             }
++//            try {
++//                document.insertString(document.getLength(), ANSI.matcher(s).replaceAll(""), (AttributeSet) null); // CraftBukkit
++//            } catch (BadLocationException badlocationexception) {
++//                ;
++//            }
```

Comment:
Use `/*` and `*/` around the block instead of commenting each individually.

Reply:
Blame IntelliJ for that one